src/Parser.cpp

```diff
@@ -0,0 +1,99 @@
+#include "../include/Parser.h"
+#include <iostream>
+#include <map>
+#include <vector>
+#include <string>
+#include <sstream>
+
+
+Parser::Parser() = default;
+
+bool Parser::checkOption(const std::string& name) {
+    if (options.find(name) != options.end()) {
+        return options[name].inCLI;
+    } else {
+        throw std::invalid_argument("Unknown option: " + name);
+    }
+}
+
+// Add a flag (boolean option)
+void Parser::addFlag(const std::string& name, const std::string& description) {
+    options[name] = Option(description, "false", true);
+}
+
+// Parse arguments
+void Parser::parse(int argc, const char* argv[]) {
+    for (int i = 1; i < argc; ++i) {
+        std::string arg = argv[i];
+        if (arg[0] == '-') {
+            // It's an option
+            std::string name = arg.substr(1);
+            if (options.find(name) != options.end()) {
+                if (options[name].isFlag) {
+                    options[name].value = "true"; // it's a flag, set to true
+                } 
+                else if (name == "search") {
+                    // Capture all words after -search until the next option
+                    std::string searchValue;
+                    while (i + 1 < argc && argv[i + 1][0] != '-') {
+                        searchValue += argv[++i];
+                        searchValue += " ";
```

Comment:
would it not make more sense to append to an array instead of making it all one big string?

Reply:
can do, that was the initial plan but string was quicker to implement